server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/core/WebSocketTransport.scala

```diff
@@ -33,14 +37,20 @@ case class WebSocketResponse(requestId: Option[RequestId], responseObject: Respo
 
 case class WebSocketError(requestId: Option[RequestId], problemDetails: ProblemDetails) extends WebSocketOutboundMessage
 
-case class StateChange(changes: Map[AccountId, TypeState]) extends WebSocketOutboundMessage {
+object PushState {
+  def from(states: State*): PushState = PushState(Hashing.sha256().hashString(states.map(_.getValue).mkString("_"), StandardCharsets.UTF_8).toString)
```

Comment:
I'm not okay with this line as it is ordering dependant:

`from(emailSate, mailboxState)` will yield a different result than `from(mailboxState, emailState)`

We do not commute and this seems like an important property to me...

I suggest instead:

```
def from(emailState: State, mailboxSatate: State)
```

No ambiguity possible, that might avoid us some bugs down the line...